Automattic/sensei
# Add settings and basic UI to category question block

### Changes proposed in this Pull Request

* Completes the MVP of the question category block design. Initial design is a bit wanting, but it at least shouldn't break things.
* Adds the question category block to the inserter.
* Creates a new helper `isQuestionEmpty` now that some blocks don't have titles.
* Edits the current behavior by allowing a value for `Number of Questions` that is greater than the number of questions currently in the category. It warns them instead.
   * This is to fix a lot of weird issues that could happen if the category question is added but then the questions removed from category.
   * I also think it makes sense, since this is a dynamic placeholder. Questions can be added and removed and the quiz needs to handle that. It might make more sense to say "Maximum Number of Questions"
   * I think we safeguard weird selection things happen on the frontend, so that shouldn't be an issue.

I'm going to wait until the base class gets merged before putting this up for review.

### Testing instructions

* Open a quiz made in the metabox editor that has a category question. Ensure it shows up correctly and can be edited.
* Add a new category question and ensure it can be reordered and saved.

### Screenshots
<img width="737" alt="Screen Shot 2021-03-17 at 2 39 04 pm" src="https://user-images.githubusercontent.com/68693/111485453-897f5180-872e-11eb-9161-09763b52ddfe.png">
<img width="276" alt="Screen Shot 2021-03-17 at 2 39 27 pm" src="https://user-images.githubusercontent.com/68693/111485522-969c4080-872e-11eb-850f-8a36612c621a.png">
<img width="285" alt="Screen Shot 2021-03-17 at 2 40 41 pm" src="https://user-images.githubusercontent.com/68693/111485703-c3505800-872e-11eb-96d8-900d10f56542.png">
